src/js/stores/ChronosStore.js

```diff
@@ -0,0 +1,143 @@
+  CHRONOS_JOBS_ERROR,
+  VISIBILITY_CHANGE
+} from '../constants/EventTypes';
+import Config from '../config/Config';
+import JobTree from '../structs/JobTree';
+import {
+  REQUEST_CHRONOS_JOBS_ERROR,
+  REQUEST_CHRONOS_JOBS_ONGOING,
+  REQUEST_CHRONOS_JOBS_SUCCESS,
+  SERVER_ACTION
+} from '../constants/ActionTypes';
+
+import VisibilityStore from './VisibilityStore';
+
+let requestInterval;
+
+function startPolling() {
+  if (!requestInterval) {
+    ChronosActions.fetchJobs();
+    requestInterval = global.setInterval(
+      ChronosActions.fetchJobs,
+      Config.getRefreshRate()
+    );
+  }
+}
+
+function stopPolling() {
+  if (requestInterval) {
+    global.clearInterval(requestInterval);
+    requestInterval = null;
+  }
+}
+
+function handleInactiveChange() {
+  let isInactive = VisibilityStore.get('isInactive');
+  if (isInactive) {
+    stopPolling();
+  }
+
+  if (!isInactive && ChronosStore.shouldPoll()) {
+    startPolling();
+  }
+}
+
+VisibilityStore.addChangeListener(VISIBILITY_CHANGE, handleInactiveChange);
+
+class ChronosStore extends EventEmitter {
+  constructor() {
+    super(...arguments);
+
+    this.data = {
+      jobTree: {id: '/', items: []}
+    };
+
+    // Handle app actions
+    this.dispatcherIndex = AppDispatcher.register(({source, action}) => {
+      if (source !== SERVER_ACTION) {
+        return false;
+      }
+      switch (action.type) {
+        case REQUEST_CHRONOS_JOBS_SUCCESS:
+          this.data.jobTree = action.data;
+          this.emit(CHRONOS_JOBS_CHANGE);
+          break;
+        case REQUEST_CHRONOS_JOBS_ONGOING:
+          break;
+        case REQUEST_CHRONOS_JOBS_ERROR:
+          this.emit(CHRONOS_JOBS_ERROR);
+          break;
+
+      }
+
+      return true;
+    })
+  }
+
+  addChangeListener(eventName, callback) {
+    this.on(eventName, callback);
+  }
+
+  removeChangeListener(eventName, callback) {
+    this.removeListener(eventName, callback);
+  }
+
+  shouldPoll() {
+    return !!this.listeners(CHRONOS_JOBS_CHANGE).length;
+  }
+
+  /**
+   * Adds the listener for the specified event
+   * @param {string} eventName
+   * @param {Function} callback
+   * @return {ChronosStore} ChronosStore instance
+   * @override
+   */
+  on(eventName, callback) {
+    super.on(eventName, callback);
+
+    // Start polling if there is at least one listener
+    if (this.shouldPoll()) {
+      startPolling();
+    }
+
+    return this;
+  }
+
+  /**
+   * Remove the specified listener for the specified event
+   * @param {string} eventName
+   * @param {Function} callback
+   * @return {ChronosStore} ChronosStore instance
+   * @override
+   */
+  removeListener(eventName, callback) {
+    super.removeListener(eventName, callback);
+
+    // Stop polling if no one is listening
+    if (!this.shouldPoll()) {
+      stopPolling();
+    }
+
+    return this;
+  }
+
+  /**
+   * @type {JobTree}
+   */
+  get jobTree() {
+    return new JobTree(this.data.jobTree);
+  }
+
+  static get storeID() {
```

Comment:
This is a `class` and we decided to use getters for stores, so this is how to access internal data. This is not inconsistent with what we have done elsewhere: https://github.com/dcos/dcos-ui/blob/master/src/js/stores/DCOSStore.js#L219